master/internal/stream/model_versions.go

```diff
@@ -258,3 +263,21 @@ func ModelVersionMakePermissionFilter(ctx context.Context, user model.User) (fun
 		}, nil
 	}
 }
+
+// ModelMakeHydrator returns a function that gets properties of a model version by
+// its id.
+func ModelVersionMakeHydrator() func(*ModelVersionMsg) (*ModelVersionMsg, error) {
+	return func(msg *ModelVersionMsg) (*ModelVersionMsg, error) {
+		var saturatedMsg ModelVersionMsg
+		query := db.Bun().NewSelect().Model(&saturatedMsg).Where("id = ?", msg.GetID()).ExcludeColumn("workspace_id")
```

Comment:
question (non-blocking): why is workspace_id being excluded from the hydrated messaged?

Reply:
nvm, I answered my own question: we're reporting an upsert that may have an old workspace & we don't want to expose any information too early

Reply:
It's obtained from the model table in the DB trigger function and send it to the backend as part of the unsaturatedMsg message.
https://github.com/determined-ai/determined/pull/9164/files#diff-b2fad5a40672d2b3108e07a06aa940de18c5a72dc6c22aa35c33c7d9b7d1e501R70